### PR TITLE
Ndev 19512

### PR DESCRIPTION
**Pull request by anushkamittal2001**

### CODE REVIEW

File: charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  labels:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: hook-succeeded
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: delete-items
+        image: bitnami/kubectl:latest
```
Comment (pns-nirmata) [inline charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml:15]: We can use our own kubectl image

File: charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          #!/bin/sh
+          set -e
+
+          # Fetch the ConfigMap
+          configmap=$(kubectl get configmap items-created-configmap -n nirmata -o json)
```
Comment (pns-nirmata) [inline charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml:23]: name of configmap can just be items-created or items-created-by-reg

File: charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          # Extract the items data
+          items=$(echo $configmap | jq -r '.data["items.json"]')
+
+          # Iterate through the items and delete them
+          echo $items | jq -c '.[]' | while read item; do
+            kind=$(echo $item | jq -r '.kind')
+            name=$(echo $item | jq -r '.name')
+
+            # Delete the item
+            kubectl delete $kind $name -n nirmata || true
```
Comment (pns-nirmata) [inline charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml:34]: There are some cluster scoped resources like clusterroles and clusterrolebindings, no namespace there. Might work even if we give namespace, but need to test.

File: charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            # Delete the ConfigMap itself
+            kubectl delete configmap custom-items-configmap -n nirmata
+          done
+        volumeMounts:
```
Comment (pns-nirmata) [inline charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml:38]: We don't need volumeMounts if we are using kubectl get cm above right?

File: charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: pre-delete-job
+  namespace: nirmata
+  labels:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: hook-succeeded
+spec:
+  template:
```
Comment (pns-nirmata) [inline charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml:10]: Also, we need to give the cluster registrator service account as the SA running the hook. Just check in kyverno-operator pre-delete hook.

File: charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml (hunk 3 above)
Comment (anushkamittal2001) [reply charts/cluster-registrator/templates/hooks/pre-delete-hook.yaml:28]: it works even if you give a namespace, I gave that cause we dont know which one is clusterscoped and which isnt at this point

**Review by pns-nirmata — APPROVED** [review]

**Pull request merged**